code/game/objects/items/devices/whistle.dm

```diff
@@ -12,33 +12,84 @@
 	var/emagged = 0
 	var/insults = 0//just in case
 
-/obj/item/device/hailer/attack_self(mob/living/carbon/user as mob)
-	if (spamcheck)
-		return
-
+/obj/item/device/hailer/proc/say_your_thing()
 	if(emagged)
-		if(insults >= 1)
-			playsound(get_turf(src), 'sound/voice/binsult.ogg', 100, 1, vary = 0)//hueheuheuheuheuheuhe
-			user.visible_message("<span class='warning'>[user]'s [name] gurgles, \"FUCK YOUR CUNT YOU SHIT EATING COCKSUCKER MAN EAT A DONG FUCKING ASS RAMMING SHITFUCK. EAT PENISES IN YOUR FUCKFACE AND SHIT OUT ABORTIONS OF FUCK AND DO A SHIT IN YOUR ASS YOU COCK FUCK SHIT MONKEY FUCK ASS WANKER FROM THE DEPTHS OF SHIT.\"</span>", \
-								"<span class='warning'>Your [name] gurgles, \"FUCK YOUR CUNT YOU SHIT EATING COCKSUCKER MAN EAT A DONG FUCKING ASS RAMMING SHITFUCK. EAT PENISES IN YOUR FUCKFACE AND SHIT OUT ABORTIONS OF FUCK AND DO A SHIT IN YOUR ASS YOU COCK FUCK SHIT MONKEY FUCK ASS WANKER FROM THE DEPTHS OF SHIT.\"</span>", \
-								"<span class='warning'>You hear the computerized voice of a security hailer: \"FUCK YOUR CUNT YOU SHIT EATING COCKSUCKER MAN EAT A DONG FUCKING ASS RAMMING SHITFUCK. EAT PENISES IN YOUR FUCKFACE AND SHIT OUT ABORTIONS OF FUCK AND DO A SHIT IN YOUR ASS YOU COCK FUCK SHIT MONKEY FUCK ASS WANKER FROM THE DEPTHS OF SHIT.\"</span>")
-			insults--
+		if(insults)
+			return "FUCK YOUR CUNT YOU SHIT EATING COCKSUCKER MAN EAT A DONG FUCKING ASS RAMMING SHITFUCK. EAT PENISES IN YOUR FUCKFACE AND SHIT OUT ABORTIONS OF FUCK AND DO A SHIT IN YOUR ASS YOU COCK FUCK SHIT MONKEY FUCK ASS WANKER FROM THE DEPTHS OF SHIT."
 		else
-			to_chat(user, "<span class='warning'>*BZZZZcuntZZZZT*</span>")
+			return "*BZZZZcuntZZZZT*"
 	else
-		playsound(get_turf(src), 'sound/voice/halt.ogg', 100, 1, vary = 0)
-		user.visible_message("<span class='warning'>[user]'s [name] rasps, \"Halt! Security!\"</span>", \
-							"<span class='warning'>Your [name] rasps, \"Halt! Security!\"</span>", \
-							"<span class='warning'>You hear the computerized voice of a security hailer: \"Halt! Security!\"</span>")
+		return "HALT! SECURITY!"
 
+/obj/item/device/hailer/proc/do_your_sound(var/mob/user)
+	if(emagged && insults)
+		playsound(get_turf(src), 'sound/voice/binsult.ogg', 100, 1, vary = 0)
+		insults--
+	else
+		playsound(get_turf(src), 'sound/voice/halt.ogg', 100, 1, vary = 0)
+	if(user)
+		var/list/bystanders = get_hearers_in_view(world.view, src)
+		flick_overlay(image('icons/mob/talk.dmi', user, "hail", MOB_LAYER+1), clients_in_moblist(bystanders), 2 SECONDS)
 	spamcheck = 1
 	spawn(20)
```

Comment:
Change this to nextuse = world.time + cooldown, and then check elsewhere if world.time > nextuse. Spawn is pretty crap for this sort of thing.